torch/csrc/jit/symbolic_script.cpp

```diff
@@ -20,6 +20,28 @@ const std::vector<std::string> functions = {
                 return grad_self, None
 
             return torch.adaptive_avg_pool2d(self, output_size), backward
+
+        def mean(self):
+            def backward(grad_output):
+                grad_self =  grad_output.expand(self.size()) / self.numel()
+                return grad_self
+
+            return torch.mean(self), backward
+
+        def var(self, unbiased: bool):
+            def backward(grad_output):
+                grad_self =  2.0 / (float(self.numel()) - float(unbiased)) * grad_output * (self - self.mean())
+                return grad_self, None
+
+            return torch.var(self, unbiased=unbiased), backward
+
+        def std(self, unbiased: bool):
```

Comment:
How did you test these? Generally to test AD formulas I try adding something incorrect to an AD formula (like + 1) and see if any of the corresponding generated JIT tests error out.

Reply:
The generated tests don't check if the graph has a `DifferentiableGraph` node, they only check for "correctness of the backward pass".  There are two cases (let's use `mean` as an example):
1) torch.mean is put into a `DifferentiableGraph` node by the `CreateAutodiffSubgraphs` pass. In this case, the generated test uses autodiff when testing the backward pass, and autodiff is tested.
2) torch.mean is not put into a `DifferentiableGraph` node because there is some bug (maybe `CreateAutodiffSubgraphs` has a bug, or the schema is wrong, or the `mean` was misspelt as `meann`. The generated test uses autograd when testing the backward pass (autodiff only gets applied to DifferentiableGraph nodes) and the torch.mean autodiff formula isn't tested as a result

Reply:
I see, that totally make sense! I will try to separate the test logic to explicitly handle the autodiff case.

Reply:
It doesn't have to happen in this PR, I think it would be good in the long term for testing